### PR TITLE
feat(interpreter): added error message for response

### PR DESCRIPTION
**Pull request by codeacme17**
Add user prompts for openai's response errors and exit in non-debug mode
Current implementation: `AuthenticationError` and `Timeout`

- AuthenticationError
  scree-shot 2023-09-11 162526

- Timeout
  scree-shot 2023-09-11 162312

### CODE REVIEW

File: interpreter/interpreter.py [hunk 1]
```diff
@@ -834,4 +838,15 @@ def messages_to_prompt(messages):
 
   def _print_welcome_message(self):
     current_version = pkg_resources.get_distribution("open-interpreter").version
-    print(f"\n Hello, Welcome to [bold white]⬤ Open Interpreter[/bold white]. (v{current_version})\n")
\ No newline at end of file
+    print(f"\n Hello, Welcome to [bold white]⬤ Open Interpreter[/bold white]. (v{current_version})\n")
```
Comment (codeacme17) [inline interpreter/interpreter.py:841]: When processing this line, a conflict will occur. Please select the code of the main branch.

**Comment by jordanbtucker** [comment]
Thanks for this. Will it still print the stack trace? That information is critical at this point in development. Once we get some proper logging it won't be as critical.

**Comment by codeacme17** [comment]
@jordanbtucker, You're right, I've reserved stack tracing for debug mode. Currently, the error will only occur when there's a response error from llm. In non-debug modes, stack traces will still appear for other errors.

**Comment by jordanbtucker** [comment]
Awesome. I think for specific errors like `openai.error.AuthenticationError` and `openai.error.Timeout`, I think it's fine to leave the stack traces out. But especially when catching generic Exceptions we should include the stack traces.

Also, probably not a good idea to call our users dummies for entering an invalid API key.

image

(I'm joking, of course 😁)

**Comment by codeacme17** [comment]
😁 If possible, could you please take a look at PR #312? I think it can be merged.

**Comment by ericrallen** [comment]
Hey there!

I'm not entirely sure this change is still valid and it would need to be updated to account for the codebase refactor, so I'm going to close it as stale for now, but feel encouraged to reopen it if you feel it's still worth pursuing and want to update it for the new codebase structure.

**Pull request closed without merging**